src/lib.rs

```diff
@@ -1,3 +1,114 @@
+//!
+//! ## State management  
+//!
+//! You will probably want your view components to have some state. You should place any state that affects
+//! you view inside a signal so that you can react to updates and update the `View`. Signals are reactive values that can be read from and written to.
+//!
+//! ### Use state to update your view
+//!
+//! To affect the layout and rendering of your component, you will need to send a state update to your component with [Id::update_state](id::Id::update_state)
+//! and then call [UpdateCx::request_layout](context::UpdateCx::request_layout) to request a layout which will cause a repaint.
+//!
+//! ## Local state
+//!
+//! To share state between components child and sibling components, you can simply pass down a signal to your children. Here's a contrived example:
+//!
+//!```rust,no_run
+//!
+//! struct Parent<V> {
+//!     id: Id,
+//!     text: ReadSignal<String>,
+//!     child: V,
+//! }
+//!
+//! // Creates a new parent view with the given child.
+//! fn parent<V>(new_child: impl FnOnce(ReadSignal<String>) -> V) -> Parent<impl View>
+//! where
+//!     V: View + 'static,
+//! {
+//!     let text = create_rw_signal(cx.scope, "World!".to_string());
+//!     // share the signal between the two children
+//!     let (id, child) = AppContext::new_id_with_child(stack(|| (text_input(text)), new_child(text.read_only()));
+//!     Parent { id, text, child }
+//! }
+//!
+//! impl<V> View for Parent<V>
+//! where
+//!     V: View,
+//! {
+//! // implementation omitted for brevity
+//! }
+//!
+//! struct Child {
+//!     id: Id,
+//!     label: Label,
+//! }
+//!
+//! /// Creates a new child view with the given state (a read only signal)
+//! fn child(text: ReadSignal<String>) -> Child {
+//!     let (id, label) = AppContext::new_id_with_child(|| label(move || format!("Hello, {}", text.get()));
+//!     Child { id, label }
+//! }
+//!
+//! impl View for Child {
+//!   // implementation omitted for brevity
+//! }
+//!
+//! // Usage
+//! fn main() {
+//!     floem::launch(parent(child));
+//! }
+//!
+//! ```
+//!
+//! ### Global state
+//!
+//! Global state can be implemented using Leptos' [provide_context](leptos_reactive::provide_context) and [use_context](leptos_reactive::use_context).
+//!
+//! # Styling
+//! Floem sizing and positioning layout system is based on the flexbox model using Taffy as the layout engine.
+//! Styles are divided into two parts:
+//! [`ComputedStyle`]: A style with definite values for most fields.  
+//!
+//! [`Style`]: A style with [`StyleValue`]s for the fields, where `Unset` falls back to the relevant
+//! field in the [`ComputedStyle`] and `Base` falls back to the underlying [`Style`] or the
+//! [`ComputedStyle`].
+//!
+//!
+//! ## Render loop and update lifecycle
+//! TBD: event -> update -> layout -> paint.
+//!
+//! potentially discuss update messages, deferred update messages, animations
+//!
+//! ## Notable invariants and tolerances
```

Comment:
Not sure if this section is all that useful. What I'm trying to do is provide some context/guarantees to build up a developer's mental model of how things are supposed to work. The browser has a bunch of these invariants (like one element can be focused at a time, there is one root element (except for in the case of an iframe), events bubble up from their source through all parents to the document, etc. 

I'd really like to build up a Floem mental model for the developer. What fundamental things are different from a browser? Event propagation, focus state, tree shape, etc.